networkx/algorithms/tree/junction_tree_algorithm.py

```diff
@@ -0,0 +1,82 @@
+r"""Function for computing a junction tree of a graph."""
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+from networkx.algorithms import moral, complete_to_chordal_graph, chordal_graph_cliques
+from itertools import combinations
+
+__all__ = ["junction_tree"]
+
+
+@not_implemented_for("multigraph", "MultiDiGraph")
+def junction_tree(G):
+    r"""Returns a junction tree of a given graph.
+
+    Notes
+    -----
+    A junction tree (or clique tree) is a tree T generated from a (un)directed graph G.
+    The tree's nodes consist of cliques and connecting sepsets of the original graph
+    respectively. The sepset of two cliques is the intersection ot the variables
+    of these cliques, e.g. the sepset of (A,B,C) and (A,C,E,F) is (A,C).
+    The junction tree algorithm consists of five steps:
+
+    1. Moralize graph
+    2. Triangulate graph
+    3. Find maximum cliques
+    4. Build graph from cliques, connecting cliques with shared
+       variables, set edge-weight to number of shared variables.
+    5. Find maximum spanning tree
+
+    https://en.wikipedia.org/wiki/Junction_tree_algorithm
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Directed graph
+
+    Returns
+    -------
+    junction_tree : NetworkX graph
+                    The corresponding junction tree of G.
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        The algorithm does not support Graph, MultiGraph and MultiDiGraph.
```

Comment:
I believe ```Graph``` should not be included in this list.

Reply:
Correct, I removed it.